pkg/custom_detectors/CUSTOM_DETECTORS.md

```diff
@@ -0,0 +1,200 @@
+# TruffleHog Custom Detector Setup Guide
+
+This guide will walk you through setting up a custom detector in TruffleHog to identify specific patterns unique to your project.
+
+## Steps to Set Up a Custom Detector
+
+1. **Create a Configuration File**:
+   - TruffleHog uses a configuration file, typically named `config.yaml`, to manage custom detector configuration.
+   - If this file doesn't exist, create it in your system.
+
+2. **Define the Custom Detector**:
+   - Open `config.yaml` with a text editor.
+   - Add a new detector under the `detectors` section.
+
+   Here's a template for a custom detector:
+
+   ```yaml
+   # config.yaml
+   detectors:
+     - name: CustomTokenDetector
+       keywords:
+         - custom
+       regex:
+         token: 'token-[a-zA-Z0-9]{16}'
+       verify:
+         - endpoint: http://localhost:8000/
+           # 'unsafe' must be set to true if the endpoint uses HTTP
+           unsafe: true
+           headers:
+             - "Authorization: super secret authorization header"
+   ```
+
+   **Explanation**:
+   - **`name`**: A unique identifier for your custom detector.
+   - **`keywords`**: An array of strings that, when found, trigger the regex search. If multiple keywords are specified, the presence of any one of them will initiate the regex search.
+   - **`regex`**: Defines the patterns to identify potential secrets. You can specify one or more named regular expressions. For a detection to be successful, each named regex must find a match.
+   - **`verify`**: An optional section to validate detected secrets. If you want to verify or unverify detected secrets, this section needs to be configured. If not configured, all detected secrets will be marked as unverified. Read [verification server examples](#verification-server-examples)
+
```

Comment:
Add explanations of the missing features: 
exclude_regexes_capture
entropy
exclude_regexes_match
exclude_words